src/platform/web/Platform.js

```diff
@@ -82,6 +82,36 @@ async function loadOlmWorker(paths) {
     return olmWorker;
 }
 
+// needed for mobile Safari which does *not* resize the viewport when the keyboard
+// shows or hides (see https://bugs.webkit.org/show_bug.cgi?id=141832)
+function adaptUIOnVisualViewportResize() {
+    if (!window.visualViewport) {
+        return;
+    }
+    window.visualViewport.addEventListener('resize', () => {
+        const sessionView = document.getElementsByClassName('SessionView')[0];
+        if (!sessionView) {
+            return;
+        }
+
+        const timeline = document.getElementsByClassName('Timeline')[0];
```

Comment:
If there's no way make this generic for all scrollable areas, better scope the selector to the container rather than the document, as the idea is that hydrogen won't touch anything outside of that root dom element (for embedding). You could also replace it by `.querySelector(".Timeline")`.